db/transforms/base.py

```diff
@@ -257,32 +257,40 @@ def map_of_output_alias_to_input_alias(self):
     def apply_to_relation(self, relation):
 
         def _get_grouping_column(col_spec):
-            preproc = col_spec.get('preproc')
-            out_alias = col_spec.get('output_alias')
-            in_alias = col_spec['input_alias']
-
-            expr = relation.columns[in_alias]
-
-            if preproc is not None:
-                expr = get_db_function_subclass_by_id(preproc).to_sa_expression(expr)
-            if out_alias is not None:
-                expr = expr.label(out_alias)
-
-            return expr
+            preproc_db_function_subclass_id = col_spec.get('preproc')
+            input_alias = col_spec['input_alias']
+            output_alias = col_spec['output_alias']
+            sa_expression = relation.columns[input_alias]
+            if preproc_db_function_subclass_id is not None:
+                sa_expression = apply_db_function_by_id(
+                    preproc_db_function_subclass_id,
+                    [sa_expression],
+                )
+            sa_expression = sa_expression.label(output_alias)
+            return sa_expression
+
+        def _get_aggregation_column(relation, col_spec):
+            input_alias = col_spec['input_alias']
+            output_alias = col_spec['output_alias']
+            agg_db_function_subclass_id = col_spec['function']
+            column_to_aggregate = relation.columns[input_alias]
+            sa_expression = column_to_aggregate
+            sa_expression = apply_db_function_by_id(
+                agg_db_function_subclass_id,
+                [sa_expression],
+            )
```

Comment:
```suggestion
            column_to_aggregate = relation.columns[input_alias]
            sa_expression = apply_db_function_by_id(
                agg_db_function_subclass_id,
                [column_to_aggregate],
            )
```
I think the extra line adds more confusion than clarity.